src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -470,10 +470,15 @@ void c_typecheck_baset::typecheck_expr_main(exprt &expr)
   {
     // already type checked
   }
+  else if(expr.id() == ID_C_spec_assigns || expr.id() == ID_target_list)
+  {
+    // already type checked
+  }
   else
   {
     error().source_location = expr.source_location();
-    error() << "unexpected expression: " << expr.pretty() << eom;
+    error() << "expression categorized improperly during parsing: "
```

Comment:
The intention was to point out that the expression's id is not handled by one of the cases we have in this typechecking function, so the ID attached to that AST node was somehow generated improperly. The expression seems to have been unexpected by the parser, so I had hoped this would clarify that phenomenon. Do you have an alternative suggestion you would prefer?

Reply:
I reverted this change.